### ISSUE TITLE
[XMaterial] - Incorrect/Inconsistent matchXMaterial(itemstack) result on 1.8.8 & 1.12.2

### ISSUE THREAD

**Issue by LOOHP** [CryptoMorin/XSeries#155]
**Description**
**1. Skeleton Skulls are matching as Dragon Heads on 1.8.8 and Pig Spawn Eggs are matching as Hoglin Spawn Eggs on 1.12.2**
Command:
```
/minecraft:replaceitem entity @p slot.hotbar.0 skull 1 0
```
For Pig Spawn Egg, it is taken from the Creative Inventory.
Code:
```java
ItemStack itemStack = player.getInventory.getItem(0);
XMaterial.matchXMaterial(itemStack);
```

**2. Inconsistency when matching different Dye items on 1.8.8**
Using the same code above, in Minecraft 1.8.8:
Bone Meal matches into Bone Meal, however
Ink Sacs matches into Black Dye

Seems like all of these incorrect/inconsistent matches involves items with 0 damage value, might have to check whether this also effects other similar items like flowers. As for the spawn egg, it might be also be due to how 1.12 uses nbt to determine the mob in contains.

**Version**
1.8.8, 1.12.2

**Comment by CryptoMorin** [comment]
What was the point of closing your old issue and opening a new one?

**Comment by LOOHP** [comment]
I discovered more issues that might or might not be related after the first issue so I thought it would be better to open a new one instead.

**Comment by CryptoMorin** [comment]
Spawn eggs use `SPAWN_EGG_META` and that's outside the scope of XMaterial. I'm not sure if I'm going to support it.
The dragon head and black dye can be fixed.

**Comment by LOOHP** [comment]
Thanks for the fix.
It's fair to say that SPAWN_EGG_META are outside the scope of XMaterial hence not supported. Although I feel like a predictable result should still be provided. Like always returning the "default material" (the one with damage 0 - i.e. PIG_SPAWN_EGG) instead of an arbitrary result like the HOGLIN_SPAWN_EGG.

**Comment by CryptoMorin** [comment]
I fixed all of them in 8.6.1
